internal/run/controller/guard.go

```diff
@@ -119,3 +123,35 @@ func (g *guard) after() error {
 
 	return nil
 }
+
+func (g *guard) changesetDiff(changesetAfter map[string]string) {
```

Comment:
I don't particularly like the interface of the method here, I think I would have rather had it return the diff and an error, and checked `g.failOnChangesDiff` at the call site. However exposing more `if`s in `after()` runs afoul with the `nestif` linter. But nevermind, I can live with this :)